src/main/java/frc/robot/subsystems/BallTransport.java

```diff
@@ -0,0 +1,118 @@
+package frc.robot.subsystems;
+
+import java.util.function.BooleanSupplier;
+import com.chopshop166.chopshoplib.SampleBuffer;
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+import com.chopshop166.chopshoplib.motors.SmartMotorController;
+import com.chopshop166.chopshoplib.sensors.IColorSensor;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.maps.RobotMap.BallTransportMap;
+
+public class BallTransport extends SmartSubsystemBase {
+    private final SmartMotorController bottomMotor;
+    private final SmartMotorController topMotor;
+    private final IColorSensor colorSensor;
+    private final BooleanSupplier laserSwitch;
+    private final Alliance allianceColor;
+
+    private boolean cargoInColorSensor = false;
+    private final double SHUNT_SPEED = 1.0;
+    private final double TRANSPORT_SPEED = 1.0;
+
+    // creates a color sample buffer.
+    private SampleBuffer<Color> colorBuffer = new SampleBuffer<Color>(2);
+
+    public BallTransport(final BallTransportMap map) {
+        this.bottomMotor = map.getBottomMotor();
+        this.topMotor = map.getTopMotor();
+        this.colorSensor = map.getColorSensor();
+        this.laserSwitch = map.getLaserSwitch();
+        this.allianceColor = DriverStation.getAlliance();
+    }
+
+    public CommandBase loadCargo() {
+        return cmd("Start Ball Transport Mechanism").onExecute(() -> {
+            if (laserSwitch.getAsBoolean()) {
+                topMotor.set(0);
+                if (cargoInColorSensor) {
+                    bottomMotor.set(0);
+                } else {
+                    bottomMotor.set(TRANSPORT_SPEED);
+                }
+            } else {
+                bottomMotor.set(TRANSPORT_SPEED);
+                topMotor.set(TRANSPORT_SPEED);
+            }
+        }).finishedWhen(() -> {
+            return cargoInColorSensor && laserSwitch.getAsBoolean();
+        }).onEnd(() -> {
+            topMotor.set(0);
+            bottomMotor.set(0);
+        });
+    }
+
+    public CommandBase shuntWrongCargo() {
+        return cmd("Shunt Wrong Cargo").onExecute(() -> {
+            switch (allianceColor) {
```

Comment:
You have duplicate logic in each case of the switch, the only difference is the colors. Why not start the command by determining which color is Our Color and which is Other Color? Then you can use the same motor logic, just comparing based on color variables.

Another issue is that `allianceColor` is set once when the subsystem is created. This is when the robot is turned on. The problem is that we don't actually get the alliance color at that point, we only get it once we're connected to the field. So this variable shouldn't be stored, it should be accessed when it's needed.